firedrake/slate/slate.py

```diff
@@ -118,6 +118,15 @@ def subdomain_data(self):
         ``{domain:{integral_type: subdomain_data}}``.
         """
 
+    @cached_property
+    def is_mixed(self):
+        """Returns `True` if the tensor has mixed arguments and `False` otherwise.
+        """
+        if any(len(arg.function_space()) > 1 for arg in self.arguments()):
+            return True
+
+        return False
```

Comment:
AKA `return any...`

```python
if condition:
    return True
return False
```

should be shortened to `return condition`.  If you really care about booleanness, then say `return bool(condition)`, but no need to here.